larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -463,25 +478,33 @@ void ThreeDOpeningAngleFeatureTool::Run(SupportVectorMachine::DoubleVector &feat
     CaloHitList threeDCaloHitList;
     LArPfoHelper::GetCaloHits(pInputPfo, TPC_3D, threeDCaloHitList);
 
-    if (threeDCaloHitList.empty())
-        return;
-
-    CartesianPointVector pointVectorStart, pointVectorEnd;
-
-    this->Divide3DCaloHitList(pAlgorithm, threeDCaloHitList, pointVectorStart, pointVectorEnd);
-    // Run the PCA analysis twice
-    CartesianVector centroidStart(0.f, 0.f, 0.f), centroidEnd(0.f, 0.f, 0.f);
-    LArPcaHelper::EigenVectors eigenVecsStart, eigenVecsEnd;
-    LArPcaHelper::EigenValues eigenValuesStart(0.f, 0.f, 0.f), eigenValuesEnd(0.f, 0.f, 0.f);
-
-    LArPcaHelper::RunPca(pointVectorStart, centroidStart, eigenValuesStart, eigenVecsStart);
-    LArPcaHelper::RunPca(pointVectorEnd, centroidEnd, eigenValuesEnd, eigenVecsEnd);
-
-    const float openingAngle(this->OpeningAngle(eigenVecsStart.at(0), eigenVecsStart.at(1), eigenValuesStart));
-    const float closingAngle(this->OpeningAngle(eigenVecsEnd.at(0), eigenVecsEnd.at(1), eigenValuesEnd));
-
-    featureVector.push_back(std::fabs(openingAngle-closingAngle));
-
+    float diffAngle(-1.f);
+    if (!threeDCaloHitList.empty())
+    {
+      CartesianPointVector pointVectorStart, pointVectorEnd;
+      this->Divide3DCaloHitList(pAlgorithm, threeDCaloHitList, pointVectorStart, pointVectorEnd);
+      if ((pointVectorStart.size()>1)&&(pointVectorEnd.size()>1)) //otherwise not able to calculate angles if only 1 point provided
```

Comment:
Prefer to add a space before e.g. `>` and `&&`

Prefer clear line between call to Divide3DCaloHitList and this if statement, and comment (if it's actually needed) above the if statement (and comment as terse as possible)